pymfl/api/SessionAPIClient.py

```diff
@@ -1,13 +1,18 @@
 from pymfl.api.MFLAPIClient import MFLAPIClient
+from pymfl.exception import MFLAPIClientException
 from pymfl.util import ConfigReader
 
+from requests import Session
+
 
 class SessionAPIClient(MFLAPIClient):
     __LOGIN_ROUTE = ConfigReader.get("api", "login_route")
 
     @classmethod
-    def get_mfl_user_id(cls, *, year: int, username: str, password: str) -> str:
+    def get_mfl_user_id(cls, *, year: int, username: str, password: str, session: Session) -> str:
         url = cls._build_route(cls._MFL_APP_BASE_URL, year, cls.__LOGIN_ROUTE)
         url = cls._add_filters(url, ("USERNAME", username), ("PASSWORD", password), ("XML", 1))
-        response = cls._post(url, as_xml=True)
-        return response.attrib["MFL_USER_ID"]  # TODO: do some error handling here
+        response = cls._post(url, session=session, as_xml=True)
```

Comment:
the `_post()` method has access to the `APIConfig` object, couldn't we retrieve the session from there instead of having to pass it in?

Reply:
I didn't see where `_post()` has access to that. By my read it looks like that's getting created after the call to `SessionAPIClient.get_mfl_user_id()`, which is what's actually calling `_post()` as far as I was able to tell.

Reply:
Now that the `APIConfig` object gets to have the session instead of just the cookie value, there's no longer a need for it to be created after the `get_mfl_user_id()` call. (And that call could probably be renamed to `login_if_necessary()` or similar.)

But I still don't see where `_post()` has access to it. I am likely missing something, but it looks to me like we'd need to pass either `APIConfig` or `Session` to `_post()`.

Reply:
`_post()` has access to the `APIConfig` object since it is a class variable of `MFLAPIClient` (`__API_CONFIG`), which is where `_post()` lives. 

You're correct in that the `APIConfig` object is being built after this call, so your implementation seems to be the way to go.